### ISSUE TITLE
resample produces TypeError: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/jwst#2527]
# Issue
From SDP run rspencer/136, running calweb_spec3 on NIS_SOSS produces the following:

<details><summary>Show log</summary>
<p>

```
strun cfgs/calwebb_spec3.cfg jw00307-a3001_20180916t185657_spec3_001_asn.json
2018-09-19 15:39:34,035 - stpipe - WARNING - /Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/photutils/background/core.py:71: AstropyDeprecationWarning: "iters" was deprecated in version 3.1 and will be removed in a future version. Use argument "maxiters" instead.
  def __init__(self, sigma_clip=SigmaClip(sigma=3., iters=5)):

2018-09-19 15:39:34,035 - stpipe - WARNING - /Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/photutils/background/core.py:114: AstropyDeprecationWarning: "iters" was deprecated in version 3.1 and will be removed in a future version. Use argument "maxiters" instead.
  def __init__(self, sigma_clip=SigmaClip(sigma=3., iters=5)):

2018-09-19 15:39:34,036 - stpipe - WARNING - /Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/photutils/background/background_2d.py:285: AstropyDeprecationWarning: "iters" was deprecated in version 3.1 and will be removed in a future version. Use argument "maxiters" instead.
  sigma_clip=SigmaClip(sigma=3., iters=10),

2018-09-19 15:39:34,124 - stpipe.Spec3Pipeline - INFO - Spec3Pipeline instance created.
2018-09-19 15:39:34,124 - stpipe.Spec3Pipeline.mrs_imatch - INFO - MRSIMatchStep instance created.
2018-09-19 15:39:34,126 - stpipe.Spec3Pipeline.outlier_detection - INFO - OutlierDetectionStep instance created.
2018-09-19 15:39:34,127 - stpipe.Spec3Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2018-09-19 15:39:34,129 - stpipe.Spec3Pipeline.cube_build - INFO - CubeBuildStep instance created.
2018-09-19 15:39:34,129 - stpipe.Spec3Pipeline.extract_1d - INFO - Extract1dStep instance created.
2018-09-19 15:39:34,129 - stpipe - INFO - Hostname: inanna2.local.stsci.edu
2018-09-19 15:39:34,130 - stpipe - INFO - OS: Darwin
2018-09-19 15:39:34,162 - stpipe.Spec3Pipeline - INFO - Step Spec3Pipeline running with args ('jw00307-a3001_20180916t185657_spec3_001_asn.json',).
2018-09-19 15:39:44,756 - stpipe.Spec3Pipeline - INFO - Prefetching reference files for dataset: 'jw00307001001_02101_00001_nis_cal.fits' reftypes = ['cubepar', 'drizpars', 'extract1d', 'resol']
2018-09-19 15:39:45,146 - stpipe.Spec3Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-09-19 15:39:45,146 - stpipe.Spec3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/niriss/jwst_niriss_drizpars_0002.fits'.
2018-09-19 15:39:45,146 - stpipe.Spec3Pipeline - INFO - Prefetch for EXTRACT1D reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/niriss/jwst_niriss_extract1d_0003.json'.
2018-09-19 15:39:45,146 - stpipe.Spec3Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-09-19 15:39:54,700 - stpipe.Spec3Pipeline - INFO - Prefetching reference files for dataset: 'jw00307002001_02101_00001_nis_cal.fits' reftypes = ['cubepar', 'drizpars', 'extract1d', 'resol']
2018-09-19 15:39:54,703 - stpipe.Spec3Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-09-19 15:39:54,703 - stpipe.Spec3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/niriss/jwst_niriss_drizpars_0002.fits'.
2018-09-19 15:39:54,703 - stpipe.Spec3Pipeline - INFO - Prefetch for EXTRACT1D reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/niriss/jwst_niriss_extract1d_0003.json'.
2018-09-19 15:39:54,703 - stpipe.Spec3Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-09-19 15:39:54,703 - stpipe.Spec3Pipeline - INFO - Starting calwebb_spec3 ...
2018-09-19 15:39:55,269 - stpipe.Spec3Pipeline.outlier_detection - INFO - Step outlier_detection running with args (<ModelContainer>,).
2018-09-19 15:39:55,272 - stpipe.Spec3Pipeline.outlier_detection - INFO - Performing outlier detection on 2 inputs
2018-09-19 15:40:30,351 - stpipe.Spec3Pipeline.outlier_detection - INFO - Saved model in jw00307001001_02101_00001_nis_a3001_crf.fits
2018-09-19 15:41:00,854 - stpipe.Spec3Pipeline.outlier_detection - INFO - Saved model in jw00307002001_02101_00001_nis_a3001_crf.fits
2018-09-19 15:41:00,855 - stpipe.Spec3Pipeline.outlier_detection - INFO - Step outlier_detection done
2018-09-19 15:41:00,892 - stpipe.Spec3Pipeline.resample_spec - INFO - Step resample_spec running with args (<ModelContainer>,).
2018-09-19 15:41:00,897 - stpipe.Spec3Pipeline.resample_spec - INFO - Drizpars reference file: /Users/eisenham/Downloads/crds-cache/references/jwst/niriss/jwst_niriss_drizpars_0002.fits
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec3Pipeline':
    'NoneType' object is not subscriptable
Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec3.py", line 132, in process
    result = self.resample_spec(result)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec_step.py", line 53, in process
    result = self._process_slit(input_models)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec_step.py", line 110, in _process_slit
    resamp = resample_spec.ResampleSpecData(input_models, **self.drizpars)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec.py", line 66, in __init__
    self.output_wcs = self.build_interpolated_output_wcs()
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec.py", line 100, in build_interpolated_output_wcs
    size_axis0 = bb[0][1] - bb[0][0] + 1
TypeError: 'NoneType' object is not subscriptable

----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/EGG-INFO/scripts/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 162, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec3.py", line 132, in process
    result = self.resample_spec(result)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec_step.py", line 53, in process
    result = self._process_slit(input_models)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec_step.py", line 110, in _process_slit
    resamp = resample_spec.ResampleSpecData(input_models, **self.drizpars)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec.py", line 66, in __init__
    self.output_wcs = self.build_interpolated_output_wcs()
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.12.0a0.dev56+g43cb952f-py3.6-macosx-10.7-x86_64.egg/jwst/resample/resample_spec.py", line 100, in build_interpolated_output_wcs
    size_axis0 = bb[0][1] - bb[0][0] + 1
TypeError: 'NoneType' object is not subscriptable
```

</p>
</details>

**Comment by stscieisenhamer** [comment]
# Resources

Data is located in `/grp/jwst/ssb/eisenhamer/jwst_issues/issue2527`

**Comment by jdavies-st** [comment]
Is SOSS data supposed be resampled?  I recall "no".

**Comment by hbushouse** [comment]
NIS_SOSS is a mode that is now switchable between TSO and non-TSO (it used to be hardwired as TSO). If it's taken in TSO mode, it definitely should NOT be going through `calwebb_spec3`. It would go through `calwebb_tso3` instead, where no resampling is involved. We should find out from the NIRISS team what kind of processing they would want for these that are taken as non-TSO. Even if they go through `calwebb_spec3`, we still can't resample them, because I don't think we have any distortion model info. At best they could simply be "stacked" as 2-D unrectified images or take the alternate route of doing 1-D extractions for the individual exposures and then combining the 1-D spectra.

**Comment by stscieisenhamer** [comment]
Right, this particular case is an association issue. Will re-label as appropriate.

**Comment by stscirij** [comment]
A NIS_SOSS exposure opens as an ImageModel and so has no .slits attribute, and since extract_2d is skipped for this mode, the WCS has no bounding box (which is the source of the error).  However, a WCS does exist giving wavelength, RA and DEC for each pixel.

**Comment by jdavies-st** [comment]
Yeah, if anything is to be done with it in `extract_2d`, `resample` or `extract_1d`, then `assign_wcs` should be computing a bounding box for it, even if that bounding box is the whole detector, as is the case for NIRCam imaging.

**Comment by jdavies-st** [comment]
And in fact, it may be a good idea to add a bounding box regardless.  Do we have the information we need to add a bounding_box in `AssignWcsStep` for NIRISS SOSS?

**Comment by philhodge** [comment]
`extract_1d` runs on NIRISS SOSS data without a bounding box by expecting the reference file to define an extraction region.

An alternative to a bounding box is for the WCS to return wavelength = NaN for pixels that are outside the region over which the WCS is defined (and that is what is currently done for many modes).  In some ways this is more useful than a bounding box because it can flag individual pixels, rather than just a rectangular region.

**Comment by stscieisenhamer** [comment]
# Implementation
This issue will deal solely with the bug that TSO data is not getting the right association type.

There is discussion here about NIS SOSS in non-TSO mode. That discussion should be opened as another issue.

- [ ] documentation
- [ ] tests
- [ ] code
- [ ] tests
- [ ] documentation
- [ ] changelog

**Comment by stscieisenhamer** [comment]
This issue is not an association issue. The exposures are marked non-TSO. 

From previous discussion, this is a calibration issue.

**Comment by hbushouse** [comment]
@nden @jdavies-st What's the current state of NIRISS SOSS mode WCS reference data? Does it not include a bounding box? If it did, is there enough additional information (transforms) to allow resampling? Of course I'm not even sure at this point if resampling is desired for SOSS.

**Comment by nden** [comment]
NIS_SOSS WCS is an objects which holds the WCS for three spectral orders. Before evaluating the WCS onehas to call `sp_wcs = niriss.niriss_soss_set_input(wcs, spectral_order)`. The `sp_wcs` has a `bounding_box`. At least this is my recollection.

**Comment by hbushouse** [comment]
Level-3 processing of non-TSO NIRISS SOSS exposures has been addressed in #4194 (see also https://jira.stsci.edu/browse/JP-985). The calwebb_spec3 pipeline skips the `resample_spec` step for these exposures and only calls `extract_1d` and `combine_1d`.

Closing.
